shell/platform/linux/fl_binary_messenger_test.cc

```diff
@@ -501,6 +501,55 @@ TEST(FlBinaryMessengerTest, AllowOverflowChannel) {
   EXPECT_TRUE(called);
 }
 
+static gboolean quit_main_loop_cb(gpointer user_data) {
+  g_main_loop_quit(static_cast<GMainLoop*>(user_data));
+  return FALSE;
+}
+
+// Checks if error returned when invoking a command on the control channel
+// are handled.
+TEST(FlBinaryMessengerTest, ControlChannelErrorResponse) {
+  g_autoptr(GMainLoop) loop = g_main_loop_new(nullptr, 0);
+  g_autoptr(FlEngine) engine = make_mock_engine();
+  FlBinaryMessenger* messenger = fl_binary_messenger_new(engine);
+
+  g_autoptr(GError) error = nullptr;
+  EXPECT_TRUE(fl_engine_start(engine, &error));
+  EXPECT_EQ(error, nullptr);
+
+  FlutterEngineProcTable* embedder_api = fl_engine_get_embedder_api(engine);
+
+  bool called = false;
+
+  FlutterEngineSendPlatformMessageFnPtr old_handler =
+      embedder_api->SendPlatformMessage;
+  embedder_api->SendPlatformMessage = MOCK_ENGINE_PROC(
+      SendPlatformMessage,
+      ([&called, old_handler](auto engine,
+                              const FlutterPlatformMessage* message) {
+        // Expect to receive a message on the "control" channel.
+        if (strcmp(message->channel, "dev.flutter/channel-buffers") != 0) {
+          return old_handler(engine, message);
+        }
+
+        called = true;
+
+        // Simulates an internal error.
+        return kInvalidArguments;
+      }));
+
+  // Register a callbalk to quit the main loop when binary messenger work ends.
+  g_idle_add(quit_main_loop_cb, loop);
```

Comment:
Can this be put inside the mock function, i.e. after the `called = true`? Although it works here it seems less clear than if it occurs after the error is detected.